src/wallet/rpcaddressbook.cpp

```diff
@@ -158,14 +186,9 @@ UniValue filteraddresses(const JSONRPCRequest &request) {
       vitMapAddressBook.push_back(it);
     }
 
-    auto comparator = sortAsc
-        ? [](const AddressBookIter &a, const AddressBookIter &b) -> bool {
-      return a->second.name.compare(b->second.name) < 0;
+    if (sort_order != SORT_NONE) {
+      std::sort(vitMapAddressBook.begin(), vitMapAddressBook.end(), comparator);
```

Comment:
nit: Given that `comparator` is just used once, I feel that defining it as a closure over the `&sort_order` reference is kinda overkill. In some sense it feels to me less intuitive than capturing `sort_order` by value, avoiding side effects.